tests/testsuite/features2.rs

```diff
@@ -1553,9 +1553,10 @@ fn edition_2021_workspace_member() {
 
     p.cargo("check").with_stderr_data(str![[r#"
 [WARNING] virtual workspace defaulting to `resolver = "1"` despite one or more workspace members being on edition 2021 which implies `resolver = "2"`
-[NOTE] to keep the current resolver, specify `workspace.resolver = "1"` in the workspace root's manifest
-[NOTE] to use the edition 2021 resolver, specify `workspace.resolver = "2"` in the workspace root's manifest
-[NOTE] for more details see https://doc.rust-lang.org/cargo/reference/resolver.html#resolver-versions
+  |
+  = [NOTE] to keep the current resolver, specify `workspace.resolver = "1"` in the workspace root's manifest
+  = [NOTE] to use the edition 2021 resolver, specify `workspace.resolver = "2"` in the workspace root's manifest
+  = [NOTE] for more details see https://doc.rust-lang.org/cargo/reference/resolver.html#resolver-versions
```

Comment:
This uses messages within a group.  So far we've instead been going separate groups for these but I think it works here because the top-level item is a warning and we have so much related content.  We've been avoiding this in other places because of the "padding" line (the line with just `|`) bloating output, because the other groups are really subordinate to a header, and it more or less keeps the UX the same while putting the code in a shape for us to experiment with what we want for the UX in the future.